site/app/controllers/admin/AdminGradeableController.php

```diff
@@ -20,6 +20,12 @@
  * @AccessControl(role="INSTRUCTOR")
  */
 class AdminGradeableController extends AbstractController {
+    /**
+     * @Route("/{_semester}/{_course}/gradeable/{gradeable_id}/homework_library", methods={"GET"})
+     */
+    private function selectFromHomeworkLibrary($gradeable_id, $semester, $course) {
```

Comment:
If you do indeed want this function to be called as a route, you need to make the access specifier public otherwise it cannot be called from the web router. Also you should type hint your gradeable_id as a Gradeable because iirc the web router, or something in between will translate the ID to an actual gradeable for you. So you can turn the $gradeable_id into Gradeable $gradeable.

Reply:
I've made the function public, and I will look into the other thing.